tests/instancetype_test.go

```diff
@@ -568,12 +562,9 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 			instancetypeRevision, err = virtClient.AppsV1().ControllerRevisions(util.NamespaceTestDefault).Get(context.Background(), newVM.Spec.Instancetype.RevisionName, metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
 
-			stashedInstancetypeSpecRevision = instancetypev1alpha1.VirtualMachineInstancetypeSpecRevision{}
-			stashedInstancetypeSpec = instancetypev1alpha1.VirtualMachineInstancetypeSpec{}
-			Expect(json.Unmarshal(instancetypeRevision.Data.Raw, &stashedInstancetypeSpecRevision)).To(Succeed())
-			Expect(stashedInstancetypeSpecRevision.APIVersion).To(Equal(updatedInstancetype.APIVersion))
-			Expect(json.Unmarshal(stashedInstancetypeSpecRevision.Spec, &stashedInstancetypeSpec)).To(Succeed())
-			Expect(stashedInstancetypeSpec).To(Equal(updatedInstancetype.Spec))
+			stashedInstancetype = &instancetypev1alpha1.VirtualMachineInstancetype{}
+			Expect(json.Unmarshal(instancetypeRevision.Data.Raw, stashedInstancetype)).To(Succeed())
+			Expect(stashedInstancetype.Spec).To(Equal(updatedInstancetype.Spec))
```

Comment:
We also spoke about adding functional test coverage of the duplicate CR detection code in `storeRevision` [1] given the issues we saw with the previous PR revision and differences testing it with fakeclients, can you add something here?

[1] https://github.com/kubevirt/kubevirt/blob/ebeed7540784cbdd05b11df99e1fc2cd7c5597fc/pkg/instancetype/instancetype.go#L254-L268

Reply:
Done